### DISCUSSION TITLE
Why there is no batchSize in RMapCache.getAll(keys) unlike RMapCache.putAll?

### DISCUSSION THREAD

**Discussion by debraj-work** [redisson/redisson#5880] [General]
I am new to Redis and I am exploring Redisson as a Java client library for use. 

Can someone let me know why there is no `batchSize` in `RMapCache.getAll(keys)` unlike `RMapCache.putAll`? Does `RMapCache.getAll()` internally do some batching while getting the keys from Redis? If not what is the recommended way to do `getAll` when there are a lot of elements in `keys` set? Do I have to manually split the `keys` into small batches and then make multiple invocations of `RMapCache.getAll()` ?

**Comment by debraj-work** [comment]
The same is asked in [StackOverflow](https://stackoverflow.com/q/78498947/785523) also.

**Comment by mrniko** [comment]
At the moment you need to split input keys into chunks manually.

**Reply by debraj-work** [reply]
Thanks for replying. 

Is there any plan to add this support in Redisson getAll in future?

Another query, in-case of [putAll](https://www.javadoc.io/doc/org.redisson/redisson/3.10.1/org/redisson/api/RMap.html#putAll-java.util.Map-int-) with batchSize, I do not see any option to pass the TTL. 

Is there a way for me to do putAll with batchSize and TTL or do I have to do this in the application side itself?
